### ISSUE TITLE
stop and refresh app

### ISSUE THREAD

**Issue by RichardHooijmaijers** [rstudio/shiny#2259]
It is possible to stop shiny when a browser tab is closed using something like `session$onSessionEnded(function() {stopApp()})` . However when this is done, it is not possible to refresh the app anymore. 

I tried an implementation using a button including `stopApp()` and javascript (`window.close()`). However

**Comment by jcheng5** [comment]
I don't really understand the question--what is it that you want, exactly? When a browser tab closes, you want to stop the Shiny app, but you still want to "refresh the app" (what does that mean when the browser tab has closed)?

**Comment by RichardHooijmaijers** [comment]
Sorry for being unclear about this..

I would want the shiny app to close when I close the browser tab. When I use `session$onSessionEnded(function() {stopApp()})`, this works but I cannot refresh the app in between (e.g. when refreshing the page, the app stops).

I looked at a workaround by adding a button that stops the app and closes the browser tab, but had no luck. So I was wondering if there is a way to stop the app and close the browser tab but still be able to refresh the page?

**Comment by vnijs** [comment]
I don't think it is possible to distinguish between a tab close and a browser refresh.

**Comment by cpenn-usgs** [comment]
Related question answered on [Stack Overflow](https://stackoverflow.com/questions/67690197/shiny-sessionreload-causing-app-shutdown-due-to-sessiononendedstopapp/73640709#73640709) that uses separate exit and reset action buttons and uses the action button value to control when `stopApp()` is run.
